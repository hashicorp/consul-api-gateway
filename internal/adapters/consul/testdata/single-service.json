{
  "Name": "single-service",
  "Hostnames": ["example.com"],
	"Rules": [{
    "Matches": [{
      "Type": 1,
      "Path": {
        "Type": 1,
        "Value": "/prefix"
      }
    }],
    "Filters": [{
      "Type": 0,
      "Header": {
        "Set": {"x-set": "1"},
        "Add": {"x-add": "2"},
        "Remove": ["x-remove"]
      }
    }],
    "Services": [{
      "Service": {
        "ConsulNamespace": "namespace",
        "Service": "service"
      },
      "Weight": 1,
      "Filters": [{
        "Type": 0,
        "Header": {
          "Set": {"x-set": "3"},
          "Add": {"x-add": "4"},
          "Remove": ["x-remove-too"]
          }
      }]
    }]
  }]
}