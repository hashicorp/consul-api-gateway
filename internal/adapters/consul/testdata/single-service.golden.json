{
  "Router": {
    "Kind": "service-router",
    "Name": "single-service",
    "Namespace": "k8s",
    "Routes": [
      {
        "Match": {
          "HTTP": {
            "PathExact": "/prefix"
          }
        },
        "Destination": {
          "Service": "service",
          "Namespace": "namespace",
          "RequestHeaders": {
            "Add": {
              "x-add": "4"
            },
            "Set": {
              "x-set": "3"
            },
            "Remove": [
              "x-remove",
              "x-remove-too"
            ]
          }
        }
      }
    ],
    "CreateIndex": 0,
    "ModifyIndex": 0
  },
  "Splitters": null
}